.github/workflows/build-nightly.yml

```diff
@@ -0,0 +1,153 @@
+name: Build (Nightly)
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - develop
+      - develop-pr-*
+    paths:
+      - "**/*.go"
+      - "**/*.c"
+      - "**/*.h"
+      - "go.mod"
+      - "go.sum"
+      - ".github/workflows/build-nightly.yml"
+      - "Makefile"
+  release:
+    types: [ prereleased ]
+  pull_request:
```

Comment:
Trigger PR is not recommended because they are not fully tested and merged into main branch.